### PR TITLE
Get Requests

### PR DESCRIPTION
**Pull request by XenithMusic**
yo i did get requests for y'all

### CODE REVIEW

File: src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionType.java [hunk 1]
```diff
@@ -119,6 +120,31 @@ public enum ScriptActionType {
...
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            conn.setRequestMethod("GET");
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream()))) {
+              for (String line; (line = reader.readLine()) != null; ) {
+                result.append(line);
+              }
+            }
+            ctx.task().context().setVariable(ctx.variable("Result").name(), new ScriptTextValue(result.toString()));
+          }catch(MalformedURLException ex){
+
```
Comment (GeorgeRNG) [inline src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionType.java:142]: A malformed URL should create an error in something like `ChatUtil.error` or preferably `OverlayManager.getInstance().add`

File: src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionType.java [hunk 2]
```diff
@@ -119,6 +120,31 @@ public enum ScriptActionType {
             ctx.value("Value")
         ))),
 
+    GET_REQUEST(builder -> builder.name("Get Webrequest")
+        .description("Makes a get request to the internet.")
```
Comment (GeorgeRNG) [inline src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionType.java:124]: With the nature of web requests, would describing how it works (sends only GET, how it fails on response, how it waits) be useful to have in the description?

File: src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionType.java [hunk 3]
```diff
@@ -119,6 +120,31 @@ public enum ScriptActionType {
...
+        .icon(Items.GRASS_BLOCK)
```
Comment (GeorgeRNG) [inline src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionType.java:125]: I think something like a comparator/repeater or command block might be more fitting for a network request, if you are not going to use a head. I understand grass block as an alternative to earth but it doesn't fit "globe"?

File: src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionType.java [hunk 4]
```diff
@@ -119,6 +120,31 @@ public enum ScriptActionType {
...
+        .arg("Result", ScriptActionArgumentType.VARIABLE)
+        .arg("Url", ScriptActionArgumentType.TEXT)
+        .category(ScriptActionCategory.VARIABLES)
+        .action(ctx -> {
+            try{
+            StringBuilder result = new StringBuilder();
+            URL url = new URL(ctx.value("Url").asText());
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            conn.setRequestMethod("GET");
```
Comment (GeorgeRNG) [inline src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionType.java:134]: Adding a method option and optional body (cannot be used in get) option would make it more flexible.

**Review by GeorgeRNG — CHANGES_REQUESTED** [review]
Unchecked GET requests

**Comment by TechStreetDev** [comment]
@XenithMusic please make the requested changes.

**Pull request merged**